src/sql/workbench/contrib/executionPlan/browser/executionPlanComparisonPropertiesView.ts

```diff
@@ -60,14 +71,31 @@ export class ExecutionPlanComparisonPropertiesView extends ExecutionPlanProperti
 			target = localize('executionPlanPropertiesEdgeOperationName', "Edge");
 		}
 
-		const titleText = localize('executionPlanComparisonPropertiesBottomOperation', "Bottom operation: {0}", target);
-		this._bottomOperationNameContainer.innerText = titleText;
-		this._bottomOperationNameContainer.title = titleText;
-		this.addDataToTable();
+		this._rightTitleText = localize('executionPlanComparisonPropertiesRightOperation', "Right operation: {0}", target);
+		this._bottomTitleText = localize('executionPlanComparisonPropertiesBottomOperation', "Bottom operation: {0}", target);
+		this._secondaryContainer.innerText = this._bottomTitleText;
+		this._secondaryContainer.title = this._bottomTitleText;
+		this.refreshPropertiesTable();
 	}
 
+	private updatePropertyContainerTitles(orientation: 'horizontal' | 'vertical'): void {
+		if (orientation === 'horizontal') {
+			this._primaryContainer.innerText = this._topTitleText;
+			this._primaryContainer.title = this._topTitleText;
+			this._secondaryContainer.innerText = this._bottomTitleText;
+			this._secondaryContainer.title = this._bottomTitleText;
+		}
+		else {
+			this._primaryContainer.innerText = this._leftTitleText;
+			this._primaryContainer.title = this._leftTitleText;
+			this._secondaryContainer.innerText = this._rightTitleText;
+			this._secondaryContainer.title = this._rightTitleText;
+		}
+
+		this.refreshPropertiesTable(orientation);
+	}
 
-	public addDataToTable() {
+	public refreshPropertiesTable(orientation: 'horizontal' | 'vertical' = 'vertical') {
```

Comment:
There's a lot of places that this is being called with the default vertical parameter which doesn't seem correct.

And in general I still don't think this should be having this parameter at all. The view already has an orientation - you should be using that directly to determine how to create the table.

Is the column name the only thing that changes when "refreshing" the table? Because you should be able to update the column names without redrawing the whole table. So I'd suggest looking into that instead to avoid the unnecessary overhead.

Reply:
In this case it's just the column name that changes.

Reply:
I would look into whether you can update the column name without refreshing the whole table - that should be possible although we may have to make some tweaks if it isn't set up to allow that easily currently.